charts/ccsm-helm/charts/tasklist/templates/_helpers.tpl

```diff
@@ -19,9 +19,25 @@ If release name contains chart name it will be used as a full name.
 {{- end -}}
 
 {{/*
-Defines labels for tasklist.
+Defines extra labels for tasklist.
+*/}}
+{{- define "tasklist.extraLabels" -}}
+app.kubernetes.io/component: tasklist
+{{- end -}}
+
+{{/*
+Define common labels for tasklist, combining the match labels and transient labels, which might change on updating
+(version depending). These labels shouldn't be used on matchLabels selector, since the selectors are immutable.
 */}}
 {{- define "tasklist.labels" -}}
 {{- template "ccsm.labels" . }}
-app.kubernetes.io/component: tasklist
+{{ template "tasklist.extraLabels" . }}
 {{- end -}}
+
+{{/*
+Defines match labels for tasklist, which are extended by sub-charts and should be used in matchLabels selectors.
+*/}}
+{{- define "tasklist.matchLabels" -}}
+{{- template "ccsm.matchLabels" . }}
+{{ template "tasklist.extraLabels" . }}
```

Comment:
The `{{ -` minus removes all whitespaces to the left (including newlines.) and vice-versa the minus `-}}` removes everything to the right. 

We include here multiple templates `matchlabels` and `extraLabels`, which have their own `{{-` etc. If we add here the `{{-` the template can't no longer rendered since the labels are added on the same line.

You can take a look at:

 * https://helm.sh/docs/chart_template_guide/control_structures/
 * https://pkg.go.dev/text/template#hdr-Text_and_spaces

Reply:
But I agree it is sometimes a bit frustrating and hard to debug and find out when to use and when not. I tried it before with the minus, but it failed to render.